compiler/luci/pass/src/RemoveRedundantTranspose.cpp

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 2020 Samsung Electronics Co., Ltd. All Rights Reserved
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "luci/Pass/RemoveRedundantTransposePass.h"
+
+#include <luci/IR/CircleNodes.h>
+#include <luci/IR/CircleOpcode.h>
+
+namespace luci
+{
+
+/// @brief Return true if first_prem[second_prem[i]] == i
+bool check_perm(const luci::CircleConst *first_perm, const luci::CircleConst *second_perm)
+{
+  assert(first_perm->rank() == 1);
+  assert(second_perm->rank() == 1);
+  assert(second_perm->size<loco::DataType::S32>() == first_perm->size<loco::DataType::S32>());
+  for (int32_t i = 0; i < static_cast<int32_t>(first_perm->size<loco::DataType::S32>()); i++)
```

Comment:
This `static_cast` seems redundant. It can be just `uint32_t`.

Reply:
If I remove that, the type of `first_perm->size<loco::DataType::S32>()` is `uint32_t`. then comparison of `int32_t` and `uint32_t` gives a error on it.

Reply:
```cxx
for (uint32_t i = 0; i < first_perm->size<loco::DataType::S32>(); i++)
```
This doesn't work?

Reply:
In that case, line below create same error.
```
if (first_perm->at<loco::DataType::S32>(second_perm->at<loco::DataType::S32>(i)) != i)
```
so I think at least one of both would be cast to another type.

Reply:
oh, got it.